cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -877,22 +878,26 @@ Result<std::shared_ptr<DatasetFactory>> ParquetDatasetFactory::Make(
 
     // Insert the path, or increase the count of row groups. It will be assumed that the
     // RowGroup of a file are ordered exactly as in the metadata file.
-    auto row_groups = &path_to_row_group_ids.insert({std::move(path), {}}).first->second;
-    row_groups->emplace_back(i);
+    auto inserted_index =
+        paths_to_index.insert({std::move(path), paths_with_row_group_ids.size()});
```

Comment:
MSVC conversion error/warning: https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/39995447/job/ldrjvpe3c7sr5sor#L1153
```suggestion
    auto inserted_index =
        paths_to_index.emplace(std::move(path), static_cast<int>(paths_with_row_group_ids.size()));
```

Reply:
Thanks.  I've applied this suggestion (didn't realize it was a suggestion when I saw it in zulip so I did it manually).